src/test/java/jenkins/scm/impl/SingleSCMSourceTest.java

```diff
@@ -61,88 +70,125 @@
 
     @Test
     public void configRoundtrip() throws Exception {
-        SingleSCMSource source = new SingleSCMSource("the-id", "the-name", new GitSCM("https://nowhere.net/something.git"));
-        SCMSourceBuilder builder = new SCMSourceBuilder(source);
-        r.assertEqualDataBoundBeans(builder, r.configRoundtrip(builder));
+        MockSCMController c = MockSCMController.create();
+        try {
+            c.createRepository("foo");
+            SingleSCMSource source =
+                    new SingleSCMSource("the-id", "the-name", new MockSCM(c, "foo",
+                            new MockSCMHead("master"), null));
+            SCMSourceBuilder builder = new SCMSourceBuilder(source);
+            r.assertEqualDataBoundBeans(builder, r.configRoundtrip(builder));
```

Comment:
Does this even work? From what I've seen `configRoundtrip` normally just returns the parameter as is. Which either means you compare between old and old data, or new and new data.